### DISCUSSION TITLE
Texture tiling

### DISCUSSION THREAD

**Discussion by danilokrulj** [google/model-viewer#3921] [Q&A]
Hi guys, has anyone tried setting texture tiling through code? I cant seem to figure out how to do this. I change the textures in material slots but the tiling isn't the same for all textures.

**Comment by AYTSistemas13** [comment]
I´ve already changed textures through code, but I don´t know what you mean when say 'tiling'.

**Reply by danilokrulj** [reply]
Its repetition of the same texture. How many times the texture is repeated on the material. In blender for example it would be UV mapping scale.

**Reply by AYTSistemas13** [reply]
In one of the models in my project this happens:
![examp](https://user-images.githubusercontent.com/80072081/200018756-7b2a9140-7b75-4089-af9e-98b76a60bc22.PNG)
This is what do you mean? if this can solve?

**Reply by danilokrulj** [reply]
Thats it! See how the texture behaves differently on different parts of model? I was looking for a way to set the scale through code but i couldnt find it. Ended up just changing the UV mapping on the model in Blender and adjusting all the textures so they fit the UV map. Although setting the tilling/repetition in code would have been much easier, it would allow using more different textures on the same model.

**Reply by AYTSistemas13** [reply]
Yes, through code I think this is not possible, I also check the documentation and not found anything about this.

**Reply by elalish** [reply]
It would be nice, and probably not too hard. PRs welcome!

**Reply by Tobieg** [reply]
Would love to have tiling function! Nice method to get super sharp textures with small file sizes.

**Reply by danilokrulj** [reply]
I wish I knew how to help. I'm still rather inexperienced to make a PR, I don't even know how that works.
